opts.go

```diff
@@ -37,7 +39,9 @@ type options struct {
 	varexp       bool
 	noParse      bool
 
-	configValueHandling configHandling
+	configValueHandling      configHandling
+	fieldValueHandling       map[string]configHandling
+	fieldValueHandlingConfig *Config
```

Comment:
How about creating a specialized type like `type fieldHandlingTable Config` (a shorter name would be welcome :) ), and provide specialized methods on this type? This makes it clear that we have a different use-case and can provide a clear API instead of us casting between configHandling and Uint when extracting the handling. WDYT?

Reply:
I like it, will do.